test/e2e/common_suite_test.go

```diff
@@ -114,15 +112,58 @@ func (tc *testCase) run() {
 				if err = client.Resources().Create(ctx, tc.pod); err != nil {
 					t.Fatal(err)
 				}
-				if err = wait.For(conditions.New(client.Resources()).PodRunning(tc.pod), wait.WithTimeout(WAIT_POD_RUNNING_TIMEOUT)); err != nil {
-					t.Fatal(err)
+				if tc.expectedPodLogString != "" {
+					if err = wait.For(conditions.New(client.Resources()).PodPhaseMatch(tc.pod, v1.PodSucceeded), wait.WithTimeout(WAIT_POD_RUNNING_TIMEOUT)); err != nil {
+						t.Fatal(err)
+					}
+				} else {
+					if err = wait.For(conditions.New(client.Resources()).PodRunning(tc.pod), wait.WithTimeout(WAIT_POD_RUNNING_TIMEOUT)); err != nil {
```

Comment:
Quick question - do the custom test images go into/end on a different state to the others e.g. nginx and that's why we need to do this branch?

Reply:
Yeah 
nginx image will be in `Running` state
our custom image will go to `Completed` state 
so made a separate branch for tracking pods with `Completed` state

Reply:
I guess my concern here is that this branch isn't really based on whether we have expected logging, but rather what the image is, so we could get into a situation where we want to check the logs of an image that stays in running. I wonder if instead of branching on this if we could update the wait to look for either running or completed and then it should be valid for all cases?

Reply:
Let me know if this solution works

- Decalred a `podState` in testCase struct
- If any pod requires different state to assess rather than `Running`, they can declare via `withCustomPodState`
- So users can declare different podState like `Succeeded`, `Pending`,`Unknown`,`Failed` and podState will dynamically add those states while setup part

Implemented above approach in updated code, please let me know your views on this approach